### DISCUSSION TITLE
BN254 twisted edwards curve = Baby Jubjub?

### DISCUSSION THREAD

**Discussion by doutv** [Consensys/gnark-crypto#521] [Q&A] [answered]
https://github.com/Consensys/gnark-crypto/blob/f483b7d9c9b49276f284441ce91af8a278c01079/ecc/bn254/twistededwards/curve.go#L54-L62

BN254's twisted edwards curve is defined here. However, the params defined in code and params defined in [eip-2494](https://eips.ethereum.org/EIPS/eip-2494) are totally different

I think they are the same curve. i.e. Baby-Jubjub

Could someone explain the differences?

**Comment by doutv** [comment]
https://github.com/Consensys/gnark-crypto/issues/87: `A=-1` to simplify group arithmetic?

**Reply by ivokub** [reply, marked as answer]
In EIP-2494 the parameters are defined as:
<img width="1106" alt="image" src="https://github.com/user-attachments/assets/e4226d2f-2ed7-46b3-98ad-270ca7332d7c">

Maybe you had a look at different curve equation parameters?

**Reply by doutv** [reply]
Thanks a lot! I ignore Reduced Twisted Edwards Form at the bottom...
